packages/sdk/src/execute.ts

```diff
@@ -41,33 +36,68 @@ const validateSigningOptions = ({ wallet, account }: NearCallSigningOptions): vo
  * @returns a result for single transactions of {@link FinalExecutionOutcome}, or void for batches
  */
 export const execute = async (
-  call: NearContractCall | NearContractCall[],
   { wallet, account }: NearCallSigningOptions,
-): Promise<void | providers.FinalExecutionOutcome> => {
+  ...calls: NearContractCall[]
+): Promise<void | providers.FinalExecutionOutcome | providers.FinalExecutionOutcome[] > => {
   validateSigningOptions({ wallet, account });
+  
+  const outcomes = await genericBatchExecute(flattenArgs(calls), wallet, account);
+  if (outcomes && outcomes.length == 1) {
+    return outcomes[0];    
+  }
+  return outcomes;
+
+};
 
+const genericBatchExecute = async (call: ContractCall[], wallet: Wallet, account: Account): Promise<void | providers.FinalExecutionOutcome[]> =>{
   if (wallet) {
-    if (call instanceof Array && call.length > 0) {
-      return await batchExecuteWithBrowserWallet(call, wallet);
-    } else {
-      return await executeWithBrowserWallet(
-        call as NearContractCall,
-        wallet,
-      );
-    }
+    return batchExecuteWithBrowserWallet(call, wallet);
   }
+  return batchExecuteWithNearAccount(call, account);
+ 
+};
 
-  if (call instanceof Array && call.length > 0) {
-    return batchExecuteWithNearAccount(call, account);
+// account call translation wrappers https://docs.near.org/tools/near-api-js/faq#how-to-send-batch-transactions
+// TODO: share batch signature with wallet selector sendAndSignTransaction when method becomes public
+
+const batchExecuteWithNearAccount = async (
+  calls: ContractCall[],
+  account: Account,
+): Promise<FinalExecutionOutcome[]> => {
+  const outcomes: FinalExecutionOutcome[] =[];
+  for (const call of calls) {
```

Comment:
In the future we might consider a parallel execution option, but for our only "real world" batch use case (storage deposit > listing) we actually want series

Reply:
I added the information that they are executed in series to the documentation. Let's for sure add the parallel execution in the future.